packit/schema.py

```diff
@@ -433,6 +433,7 @@ class UserConfigSchema(Schema):
     command_handler_image_reference = fields.String()
     command_handler_k8s_namespace = fields.String()
     kerberos_realm = fields.String()
+    package_config_path = fields.String(default=None)
```

Comment:
I see the implementation is really easy by using this, but do we expect this to be used in the config for real? Isn't this a bit misleading that it is in the config schema but never used properly?

Reply:
This could be used by users to set some other path for package configs than the default hard-coded options. Though I have to agree, that just by itself, its usefulness is quite limited. Given this, I would also propose to remove it for now.

Reply:
I think, that it may be handy to specify the path to package-config file in packit-config [1]. This also allows to specify the path in `packit-service`.

AFAIK, removing it from config schema would mean to reimplement the way the packit-config (`Config`) is passed by `click`. Currently the `Config` object is passed as [`ctx.obj`](https://github.com/packit/packit/blob/9e2be24c2cbf14fd981bf8ed9673ca7ea5fbace2/packit/cli/packit_base.py#L80). After removal of `package_config_path` from `Config` object, we would have to create a new class to encapsulate the `Confg` object and other command-line options (like `package_config_path`),  in order to pass them in `click`'s `ctx.obj`. This would lead to further changes in every packit command (`build`, `copr-build`, ...).

The `debug` option (and `dry-run` until its recent removal) is also part of the schema and I believe, that it is meant to be used from command-line.

---
[1] I found a bug in [my implementation](https://github.com/packit/packit/blob/9e2be24c2cbf14fd981bf8ed9673ca7ea5fbace2/packit/cli/packit_base.py#L79) that needs to be fixed in order to make it work as intended.

Reply:
Okay, this makes sense. I'm fine with keeping it in packit config. Thanks for the explanation!

Reply:
> After removal of package_config_path from Config object, we would have to create a new class to encapsulate the Confg object and other command-line options (like package_config_path), in order to pass them in click's ctx.obj.

Just an idea: can we add an attribute to the `Config` class that is not part of the schema?

> The `debug` option (and `dry-run` until its recent removal) is also part of the schema and I believe, that it is meant to be used from command-line.

This one is handy to have -- e.g. I have it in my user config set ...;)

----

But I am not against. That was just to compare this approach to potential alternatives.

Reply:
> > After removal of package_config_path from Config object, we would have to create a new class to encapsulate the Confg object and other command-line options (like package_config_path), in order to pass them in click's ctx.obj.
> 
> Just an idea: can we add an attribute to the `Config` class that is not part of the schema?

I think we cannot: 
https://github.com/packit/packit/blob/9e2be24c2cbf14fd981bf8ed9673ca7ea5fbace2/packit/config/config.py#L142-L150

Reply:
Well, we probably could  remove it from schema, because so far we use it only for reading the config. But then the `UserConfigSchema` wouldn't reflect the `Config` object.

This could cause problems in future, eg. if we wanted to serialize config to dict and then get it back, then ```config2.get_from_dict(config.serialize_to_dict())``` would result in `config2  != config`.

Reply:
Makes sense. Looks like it will cause more problems than benefits.